src/lib.rs

```diff
@@ -79,6 +80,11 @@ const KEEPALIVE_IDLE_DURATION: Duration = Duration::from_secs(65);
 const KEEPALIVE_INTERVAL: Duration = Duration::from_secs(65);
 const KEEPALIVE_RETRIES: u32 = 1;
 
+// Rate-limiting configuration
+const RATE_LIMIT_MAX_TOKENS: u32 = 300;
+const RATE_LIMIT_INTERVAL: Duration = Duration::from_secs(1);
+const RATE_LIMIT_REFILL_RATE: u32 = 100;
```

Comment:
Looks like this is 100 req/s per IP? Seems really high. What about 2 req/s with burst of 100?

Plus I think there should be a second rate limit for the project, e.g. 1k/s.

```suggestion
const RATE_LIMIT_MAX_TOKENS: u32 = 100;
const RATE_LIMIT_INTERVAL: Duration = Duration::from_secs(1);
const RATE_LIMIT_REFILL_RATE: u32 = 2;
```

Reply:
Makes sense, probably it's too high.
I've moved the configuration to env variables in 09d67304f6a1544938fb4d1ad964d62a64e7b172 so we don't need to rebuild everything just to change it. 
Let's start with 100 burst and 2/sec and then I'll check the metric from e8a5cc486a69ae6ac11a70f25a0cd6c9ffc44ce9 how many are affected.

Reply:
> Plus I think there should be a second rate limit for the project, e.g. 1k/s.

Speaking about project ID limiting I think this should be a follow-up PR. Let's ship this incrementally.